File: DSA/Animesh_jha_590016813/Day_66_question_2.py
def max_profit(prices):
    min_price = float('inf')
    max_profit = 0
    
    for price in prices:
        min_price = min(min_price, price)
        max_profit = max(max_profit, price - min_price)
    
    return max_profit

print(max_profit([7, 1, 5, 3, 6, 4])) 
print(max_profit([7, 6, 4, 3, 1]))  

File: DSA/GEHNA_KHANNA_590013140/day66question2.c
#include <stdio.h>

int main(void) {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    if (n <= 0) { printf("0\n"); return 0; }

    int price, minPrice, maxProfit = 0;
    scanf("%d", &price);
    minPrice = price;

    for (int i = 1; i < n; i++) {
        scanf("%d", &price);
        if (price - minPrice > maxProfit) maxProfit = price - minPrice;
        if (price < minPrice) minPrice = price;
    }
    printf("%d\n", maxProfit);
    return 0;
}
